### ISSUE TITLE
[Policy Assistant] user experience for specifying pods/traffic

### ISSUE THREAD

**Issue by huntergregory** [kubernetes-sigs/network-policy-api#220]
Parent issue: #150 

## TL;DR

Write go code to get a Deployment/DaemonSet from a cluster and create a corresponding `TrafficPeer` (see struct referenced below).

## Overview

Our KubeCon demo ([video](https://youtu.be/riSv0g-TNtI?si=6gxDON4jK2xf4n-e&t=1449), [slides](https://static.sched.com/hosted_files/kccnceu2024/bc/KubeConEU24_NetworkPolicy_API.pptx.pdf), [code](https://github.com/huntergregory/network-policy-api/tree/kubecon/cmd/policy-assistant/kubecon)) used hard-coded Pods and traffic (Pod a to Pod b on port 80 and 81).
![image](https://github.com/kubernetes-sigs/network-policy-api/assets/42728408/688b4f52-82b4-44b8-8490-5e54e6739093)

Of course, a user should be able to specify their own Pods/traffic. Ideally, a user can:
- Quickly/easily specify traffic of interest.
- Reference Pods in a Kubernetes-native way.

## Ideas

A helpful starting point is Matt's [Traffic struct](https://github.com/kubernetes-sigs/network-policy-api/blob/669dfbc72245a2c7c6f3a3dd5c6188b7af9f8014/cmd/policy-assistant/pkg/matcher/traffic.go#L13). Here's a snippet of this [example json](https://github.com/kubernetes-sigs/network-policy-api/blob/main/cmd/policy-assistant/examples/traffic.json):
```json
{
  "Source": {
    "IP": "8.8.8.8"
  },
  "Destination": {
    "Internal": {
      "PodLabels": {"pod": "a"},
      "NamespaceLabels": {"ns":  "y"},
      "Namespace": "y"
      // we will also need NodeLabels for AdminNetworkPolicy's node selector
    },
    "IP": "192.168.1.10"
  },
  "Protocol": "TCP",
  "ResolvedPort": 80,
  "ResolvedPortName": "serve-80-tcp"
}
```

**_It would be nice if a user could instead reference a Pod/Deployment/DaemonSet_**, and then Policy Assistant queries someone's cluster to fill in:
- pod labels
- namespace labels
- node labels
- IP (or IPs of a Deployment, for instance)

We could start by building go code to convert a Deployment or DaemonSet to a `TrafficPeer` for a user's Kubernetes cluster.

### Separate Discussion: what should be the CLI Experience?

For the command in the screenshot above, should a user specify that configuration in JSON like so?

```bash
policy-assistant analyze --mode walkthrough --traffic-file traffic.json
```

where `traffic.json` is like:
```json
{
    "Traffic": [
        {
            "From": "ns-dev/deployment/frontend",
            "To": "ns-dev/daemonset/backend",
            "Protocol": "TCP",
            "PortName": "serve-80-tcp"
        },
        {
            "From": "10.0.0.5",
            "To": "ns-dev/daemonset/backend",
            "Protocol": "UDP",
            "Port": "81"
        },
    ]
}
```

**Comment by gabrielggg** [comment]
Hey @huntergregory , i think we can close this issue!

**Comment by huntergregory** [comment]
Thanks for the reminder here @gabrielggg. Have we tested the end to end experience of specifying pods/traffic in the cli ? Based on the foundations you set in #227

**Comment by gabrielggg** [comment]
> Thanks for the reminder here @gabrielggg. Have we tested the end to end experience of specifying pods/traffic in the cli ? Based on the foundations you set in #227

Hey @huntergregory , you are right, that part is missing to close this issue, i think that as soon as this PR (https://github.com/kubernetes-sigs/network-policy-api/pull/245) gets merged, i will begin work on a new PR for achieving the full end to end experience!

**Comment by huntergregory** [comment]
Hey @gabrielggg , just merged #245 ! If you're still interested, I'm excited to see what you build! We're looking to cut a release for policy assistant by about Nov 5 if you could possibly raise a pr before then

**Comment by gabrielggg** [comment]
i'm working on this, i think i will make it on time! @huntergregory

**Comment by huntergregory** [comment]
Hey @gabrielggg how's this change going?

**Comment by huntergregory** [comment]
Addressed in #264
